src/ethereum/api/Bridge.ts

```diff
@@ -3,16 +3,137 @@ import { CollectionAttributes, Collection } from '../../Collection'
 import { ItemAttributes, Item, FullItem } from '../../Item'
 import { fromUnixTimestamp } from '../../utils/parse'
 import { buildTPItemURN } from '../../Item/utils'
-import { isTPCollection } from '../../Collection/utils'
-import { ItemFragment, CollectionFragment } from './fragments'
+import {
+  ItemFragment,
+  CollectionFragment,
+  ThirdPartyItemFragment,
+} from './fragments'
 import { collectionAPI } from './collection'
-import { Wearable } from './peer'
+import { peerAPI, Wearable } from './peer'
+import { thirdPartyAPI } from './thirdParty'
+import { isTPCollection } from '../../Collection/utils'
 
 export class Bridge {
+  static async consolidateTPCollections(
+    dbCollections: CollectionAttributes[]
+  ): Promise<CollectionAttributes[]> {
+    const collections: CollectionAttributes[] = []
+
+    for (const dbCollection of dbCollections) {
+      let fullCollection: CollectionAttributes = { ...dbCollection }
+
+      if (isTPCollection(dbCollection)) {
+        const lastItem = await thirdPartyAPI.fetchLastItem(
+          dbCollection.third_party_id,
+          dbCollection.urn_suffix
+        )
+        if (lastItem) {
+          fullCollection = Bridge.mergeTPCollection(dbCollection, lastItem)
+        }
+      }
+
+      collections.push(fullCollection)
+    }
+    return collections
+  }
+
+  static async consolidateTPItems(
+    dbItems: ItemAttributes[],
+    remoteItems: ThirdPartyItemFragment[]
+  ): Promise<FullItem[]> {
+    const dbTPItemIds = dbItems.map((item) => item.id)
+    const dbTPCollections = await Collection.findByIds(dbTPItemIds)
+
+    const itemsByURN: Record<
+      string,
+      { item: ItemAttributes; remoteItem?: ThirdPartyItemFragment }
+    > = {}
+
+    const tpItemURNs = dbItems.map((item) => {
+      const collection = dbTPCollections.find(
+        (collection) => collection.id === item.collection_id
+      )
+      if (!collection || !isTPCollection(collection)) {
+        throw new Error(`Could not find a valid collection for item ${item.id}`)
+      }
+
+      const urn = buildTPItemURN(
+        collection.third_party_id,
+        collection.urn_suffix,
+        item.urn_suffix!
+      )
+
+      const remoteItem = remoteItems.find(
+        (remoteItem) => remoteItem.urn === urn
+      )
+
+      itemsByURN[urn] = { item, remoteItem }
+
+      return urn
+    })
+
+    const tpCatalystItems = await peerAPI.fetchWearables(tpItemURNs)
+    const fullItems: FullItem[] = []
+
+    for (const urn in itemsByURN) {
```

Comment:
I have no strong feelings either way. I usually favor `for(in/of)` for simplicity's sake but it's just a personal taste kind of thing.
Reading
`Object.keys(itemsByURN).map(() => {})`
_feels_ a more cumbersome that
`for(const urn in itemsByURN) {}`

That said, in this context, it does simplify the return value of the function so I'll change it!

Reply:
Oh, I forgot that the `itemsByURN` was a map. I don't have a strict preference on going through map keys (although I usually use the `Object.keys` one), but it might be better to use a map if we're just pushing into an array.